src/bvals/fc/bvals_fc.hpp

```diff
@@ -72,12 +72,14 @@ class FaceCenteredBoundaryVariable : public BoundaryVariable {
 #endif
 
   // BoundaryBuffer:
-  int LoadBoundaryBufferSameLevel(Real *buf, const NeighborBlock &nb) override;
-  void SetBoundarySameLevel(Real *buf, const NeighborBlock &nb) override;
-  int LoadBoundaryBufferToCoarser(Real *buf, const NeighborBlock &nb) override;
-  int LoadBoundaryBufferToFiner(Real *buf, const NeighborBlock &nb) override;
-  void SetBoundaryFromCoarser(Real *buf, const NeighborBlock &nb) override;
-  void SetBoundaryFromFiner(Real *buf, const NeighborBlock &nb) override;
+  int LoadBoundaryBufferSameLevel(ParArray1D<Real> &buf,
+                                  const NeighborBlock &nb) override;
+  void SetBoundarySameLevel(ParArray1D<Real> &buf, const NeighborBlock &nb) override;
+  int LoadBoundaryBufferToCoarser(ParArray1D<Real> &buf,
+                                  const NeighborBlock &nb) override;
+  int LoadBoundaryBufferToFiner(ParArray1D<Real> &buf, const NeighborBlock &nb) override;
+  void SetBoundaryFromCoarser(ParArray1D<Real> &buf, const NeighborBlock &nb) override;
+  void SetBoundaryFromFiner(ParArray1D<Real> &buf, const NeighborBlock &nb) override;
```

Comment:
I would recommend using `final` over `override` in this case. There are sometimes performance benefits:

https://devblogs.microsoft.com/cppblog/the-performance-benefits-of-final-classes/

Reply:
Good point. I'll change that.